bokeh-demo/examples/landing_page/main.py

```diff
@@ -1,25 +1,249 @@
+        self.min_age_last_exam_slider = self._init_min_age_last_exam_slider()
+        self.min_n_screenings_slider = self._init_min_n_screenings_slider()
 
-# currently saving file on every change
-text.on_change("value", save_text)
+        self.save_button = self._init_save_button()
+        self.person_counter = Div(text=f"Number of people: {self._n_people()}")
+        self.go_button = self._init_go_button()
 
-for root_element in (text, go_button):
+    def get_roots(self) -> list[Model]:
+        return [
+            self.age_slider,
+            self.min_age_last_exam_slider,
+            self.min_n_screenings_slider,
+            self.checkbox_buttons,
+            self.person_counter,
+            self.save_button,
+            self.go_button,
+        ]
+
+    def reset_pid_list(self) -> None:
+        self.pid_list = self.person_df.index.to_list()
+
+    def _n_people(self) -> int:
+        return len(self.pid_list)
+
+    def _add_HR_screening_indicator(self) -> None:
+        for exam_type, df in self.exams_df.groupby(self.column_names.exam_type):
+            if exam_type == "HPV":
+                continue
+            column_name = f"high_risk_{exam_type.lower()}"
+            self.person_df.loc[
+                df[df[self.column_names.risk] > 2][self.column_names.PID].unique(),
+                column_name,
+            ] = True
+            self.person_df[column_name].fillna(False, inplace=True)
+
+    def apply_filters_and_save(self) -> None:
+        self.reset_pid_list()
+        age_pids = self.age_slider_pids()
+        button_pids = self.checkbox_button_pids()
+        min_age_last_exam_pids = self.min_age_last_exam_slider_pids()
+        min_n_screenings_pids = self.min_n_screenings_slider_pids()
+
+        self.pid_list = list(
+            map(
+                int,
+                list(
+                    set(age_pids)
+                    & set(button_pids)
+                    & set(min_age_last_exam_pids)
+                    & set(min_n_screenings_pids)
+                ),
+            )
+        )
+        self.person_counter.text = f"Number of people: {self._n_people()}"
+        self.save_pids()
+
+    def _init_go_button(self) -> Button:
+        go_button = Button(label="Go")
+        go_button.js_on_click(CustomJS(code="window.location.href = '/pilot'"))
+        return go_button
+
+    def _init_save_button(self) -> Button:
+        save_button = Button(label="Save")
+        save_button.on_click(self.apply_filters_and_save)
+        return save_button
+
+    def _init_age_slider(self) -> RangeSlider:
+        birthyears = self.person_df[self.column_names.birthdate].dt.year
+        start, end = birthyears.min(), birthyears.max()
+        age_slider = RangeSlider(
+            name="age_slider",
+            start=start,
+            end=end,
+            value=(start, end),
+            step=1,
+            title="Birth Year",
+        )
+        return age_slider
```

Comment:
```suggestion
        return RangeSlider(
            name="age_slider",
            start=start,
            end=end,
            value=(start, end),
            step=1,
            title="Birth Year",
        )
```

Reply:
According to https://docs.sourcery.ai/Reference/Python/Default-Rules/inline-immediately-returned-variable/. Up to you if you want this or not.